src/utils/conversion.ts

```diff
@@ -4067,3 +4074,189 @@ export function inputStatTypeToMeshStatType(
   }
   return statisticsOpTypeToStatType({ op, claimIssuer }, context);
 }
+
+/**
+ * @hidden
+ */
+export function metadataSpecToMeshMetadataSpec(
+  specs: MetadataSpec,
+  context: Context
+): PolymeshPrimitivesAssetMetadataAssetMetadataSpec {
+  const { url, description, typeDef } = specs;
+
+  const {
+    polymeshApi: {
+      consts: {
+        asset: { assetMetadataTypeDefMaxLength },
+      },
+    },
+  } = context;
+
+  const metadataTypeDefMaxLength = u32ToBigNumber(assetMetadataTypeDefMaxLength);
+
+  if (typeDef && metadataTypeDefMaxLength.lt(typeDef.length)) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: '"typeDef" length exceeded for given Asset Metadata spec',
+      data: {
+        maxLength: metadataTypeDefMaxLength,
+      },
+    });
+  }
+
+  return context.createType('PolymeshPrimitivesAssetMetadataAssetMetadataSpec', {
+    url: optionize(stringToBytes)(url, context),
+    description: optionize(stringToBytes)(description, context),
+    typeDef: optionize(stringToBytes)(typeDef, context),
+  });
+}
+
+/**
+ * @hidden
+ */
+export function meshMetadataSpecToMetadataSpec(
+  rawSpecs?: Option<PolymeshPrimitivesAssetMetadataAssetMetadataSpec>
```

Comment:
Do we need `?` and `Option`? I would have though the Option covered the undefined case.

Reply:
We need `?`, otherwise we would have to handle the undefined case at the respective places from where this is invoked.